pkg/operator/certrotation/metadata.go

```diff
@@ -1,36 +1,40 @@
 package certrotation
 
 import (
+	"fmt"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func ensureOwnerRefAndTLSAnnotations(secret *corev1.Secret, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) bool {
-	needsMetadataUpdate := false
+func ensureOwnerRefAndTLSAnnotationsForSecret(secret *corev1.Secret, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) string {
+	updateDetails := ""
 	// no ownerReference set
-	if owner != nil {
-		needsMetadataUpdate = ensureOwnerReference(&secret.ObjectMeta, owner)
+	if owner != nil && ensureOwnerReference(&secret.ObjectMeta, owner) {
+		updateDetails = fmt.Sprintf("owner reference updated to %#v", owner)
 	}
 	// ownership annotations not set
-	return additionalAnnotations.EnsureTLSMetadataUpdate(&secret.ObjectMeta) || needsMetadataUpdate
+	if additionalAnnotations.EnsureTLSMetadataUpdate(&secret.ObjectMeta) {
+		if len(updateDetails) > 0 {
+			updateDetails += ", "
+		}
+		updateDetails += fmt.Sprintf("annotations set to %#v", additionalAnnotations)
+	}
+	return updateDetails
 }
 
-func ensureSecretTLSTypeSet(secret *corev1.Secret) bool {
-	// Existing secret not found - no need to update metadata (will be done by needNewSigningCertKeyPair / NeedNewTargetCertKeyPair)
-	if len(secret.ResourceVersion) == 0 {
-		return false
+func ensureOwnerRefAndTLSAnnotationsForConfigMap(configMap *corev1.ConfigMap, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations) string {
```

Comment:
why not to have `ensureOwnerRefAndTLSAnnotations((meta *metav1.ObjectMeta, owner *metav1.OwnerReference, additionalAnnotations AdditionalAnnotations)` which could be used for both `cm` and `secrets`?